lib/services/wallet_connect_service.dart

```diff
@@ -334,19 +342,45 @@ class WalletConnectService {
             event =
                 RequestSendQubicEvent.fromMap(args, topic, sessionRequest.id);
             event.validateOrThrow();
-
-            if (web3Wallet!.getActiveSessions().containsKey(topic)) {
-              event.setPairingMetadata(
-                  web3Wallet!.getActiveSessions()[topic]!.peer.metadata);
-            } else {
-              throw "Session not found";
+            validateAndSetSession(topic, event);
+            return web3Wallet!.respondSessionRequest(
+                topic: topic,
+                response: JsonRpcResponse(
+                    id: sessionRequest.id,
+                    result: await sendQubicHandler!(event)));
+          } catch (e) {
+            if (e is JsonRpcError) {
+              rethrow;
             }
+            throw JsonRpcError(code: -2, message: e.toString());
+          }
+        });
 
+    web3Wallet!.registerRequestHandler(
+        chainId: Config.walletConnectChainId,
+        method: WcMethods.wSendTransaction,
+        handler: (topic, args) async {
+          final sessionRequest = web3Wallet!.pendingRequests
+              .getAll()
+              .where((e) =>
+                  e.method == WcMethods.wSendTransaction && e.topic == topic)
+              .last;
+
+          late RequestSendTransactionEvent event;
+
+          if (sendTransactionHandler == null) {
+            throw "sendTransactionHandler is not set";
+          }
+          try {
+            event = RequestSendTransactionEvent.fromMap(
+                args, topic, sessionRequest.id);
+            event.validateOrThrow();
+            validateAndSetSession(topic, event);
             return web3Wallet!.respondSessionRequest(
                 topic: topic,
                 response: JsonRpcResponse(
                     id: sessionRequest.id,
-                    result: await sendQubicHandler!(event)));
+                    result: await sendTransactionHandler!(event)));
           } catch (e) {
             if (e is JsonRpcError) {
```

Comment:
@ahmed-tarek-salem  I was checking documentation and examples and it seems the way we are throwing the error is not correct.
web3Wallet!.respondSessionRequest( can be also executed with JsonRpcError object.

Reply:
![image](https://github.com/user-attachments/assets/bd11f64e-4089-4d24-bf41-f8aa2befebb4)

I'm also curious if we are supporting deep links for events and the redirect after the user responds to this